src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c

```diff
@@ -21,47 +48,601 @@ const ActorInit Bg_Po_Event_InitVars = {
+            }
+            if (Actor_SpawnAsChild(&globalCtx->actorCtx, &this->dyna.actor, globalCtx, ACTOR_BG_PO_EVENT,
+                                   paintingsX[phi_t2], paintingsY[this->actorIndex], paintingsZ[this->actorIndex], 0,
+                                   this->dyna.actor.shape.rot.y + 0x8000, 0,
+                                   ((this->actorIndex + 1) << 0xC) + (this->actorType << 8) +
+                                       this->dyna.actor.params) == 0) {
+                Actor_Kill(&this->dyna.actor);
+                return;
+            }
+            if (this->actorIndex == 0) {
+                if (this->dyna.actor.child->child == NULL) {
+                    Actor_Kill(&this->dyna.actor);
+                    return;
+                }
+                this->dyna.actor.parent = this->dyna.actor.child->child;
+                this->dyna.actor.child->child->child = &this->dyna.actor;
+            }
+        }
+    }
+    this->timer = 0;
+    if (this->actorType == 4) {
+        puzzleState = 0;
+        this->actionFunc = &BgPoEvent_AmyWait;
+    } else {
+        puzzleState = (s32)(Math_Rand_ZeroOne() * 3.0f) % 3;
+        this->actionFunc = &BgPoEvent_PaintingEmpty;
+    }
+}
+
+void BgPoEvent_InitBlocks(BgPoEvent* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->dyna.actor;
+    s32 local_c = 0;
+    s32 dummy;
+
+    thisx->flags |= 0x30;
+    DynaPolyInfo_Alloc(&D_06007860, &local_c);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, local_c);
+    if ((this->actorType == 0) && (this->actorIndex != 3)) {
+        if (Actor_SpawnAsChild(&globalCtx->actorCtx, thisx, globalCtx, ACTOR_BG_PO_EVENT, blocksX[this->actorIndex],
+                               thisx->posRot.pos.y, blocksZ[this->actorIndex], 0, thisx->shape.rot.y,
+                               thisx->shape.rot.z - 0x4000,
+                               ((this->actorIndex + 1) << 0xC) + (this->actorType << 8) + thisx->params) == NULL) {
+            Actor_Kill(thisx);
+            return;
+        }
+        if (this->actorIndex == 0) {
+            if (thisx->child->child == 0) {
+                Actor_Kill(&this->dyna.actor);
+                return;
+            }
+            if (thisx->child->child->child == 0) {
+                Actor_Kill(thisx);
+                Actor_Kill(thisx->child);
+                return;
+            }
+            thisx->parent = thisx->child->child->child;
+            thisx->child->child->child->child = thisx;
+        }
+    }
+    thisx->posRot.pos.y = 833.0f;
+    thisx->groundY = func_8003C9A4(&globalCtx->colCtx, &thisx->floorPoly, &dummy, thisx, &thisx->posRot.pos);
+    this->actionFunc = BgPoEvent_BlockWait;
+}
+
+void BgPoEvent_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgPoEvent* this = THIS;
+    ColliderTris* collider;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->actorType = (thisx->params >> 8) & 0xF;
```

Comment:
can you avoid using thisx?

Reply:
For some reason, at most one of those three params statements can use this->dyna.actor. If two or more do, it fails to match. I suppose I could leave the two thisx in the actortype and actorindex statements and switch the rest to this->dyna.actor